ssl/t1_lib.c

```diff
@@ -2897,11 +2899,27 @@ static int sig_cb(const char *elem, int len, void *arg)
      * in the table.
      */
     if (p == NULL) {
-        for (i = 0, s = sigalg_lookup_tbl; i < OSSL_NELEM(sigalg_lookup_tbl);
-             i++, s++) {
-            if (s->name != NULL && strcmp(etmp, s->name) == 0) {
-                sarg->sigalgs[sarg->sigalgcnt++] = s->sigalg;
-                break;
+        /* Load provider sigalgs */
+        if (sarg->ctx && ssl_load_sigalgs(sarg->ctx)) {
+            /* Check if a provider supports the sigalg */
+            for (i = 0; i < sarg->ctx->sigalg_list_len; i++) {
+                if (sarg->ctx->sigalg_list[i].sigalg_name != NULL
+                    && strcmp(etmp,
+                              sarg->ctx->sigalg_list[i].sigalg_name) == 0) {
+                    sarg->sigalgs[sarg->sigalgcnt++] =
+                            sarg->ctx->sigalg_list[i].code_point;
+                    break;
```

Comment:
What if `sarg->ctx->sigalg_list_len > OSSL_NELEM(sigalg_lookup_tbl)` and `i==OSSL_NELEM(sigalg_lookup_tbl)` on this `break`? I'd argue it'd be prudent to have tests for these (and other) cases.

Reply:
Nice catch, I this stems from my reuse of the `i` var for both for loops. I'll add another incrementation var so each loop uses it's own

Reply:
after a closer read of my code changes inside this block I found a few edge cases for how it could traverse the if blocks. So instead I've moved the `OSSL_NELEM` check inside the if block above it alongside the code previously in this section then separated the new if check into two steps to remove any unexpected edge cases due to fallen calls or for loops.

Reply:
A test would arguably still be better than relying on visual inspection to find edge cases.